ansible_ai_connect/ai/api/views.py

```diff
@@ -770,10 +841,83 @@ def post(self, request) -> Response:
                 "format": "plaintext",
                 "generationId": generation_id,
             }
+
+        except ModelTimeoutError as e:
+            exception = e
+            logger.warn(
+                f"model timed out after {settings.ANSIBLE_AI_MODEL_MESH_API_TIMEOUT} seconds"
+                f" for playbook generation {generation_id}"
+            )
+            raise ModelTimeoutException(cause=e)
+
+        except WcaBadRequest as e:
+            exception = e
```

Comment:
The main goal in pushing this out the quick-and-dirty way was to get some extra logging in the case where we're getting 400 errors from WCA. The current handling doesn't log the response body anywhere. Shall we add some extra handling like we have here? https://github.com/ansible/ansible-ai-connect-service/blob/main/ansible_ai_connect/ai/api/model_client/wca_utils.py#L161-L163

Reply:
Indeed.

I now call the chain of response handlers that perform the exact logging you link.

https://github.com/ansible/ansible-ai-connect-service/pull/1157/files#diff-052b0c3ff4c4ad8cbf30bbdaa0f88e9157b5e33a94c6ea85d38257f7010b576eR495-R496

https://github.com/ansible/ansible-ai-connect-service/pull/1157/files#diff-052b0c3ff4c4ad8cbf30bbdaa0f88e9157b5e33a94c6ea85d38257f7010b576eR525-R526

Reply:
400s aren't getting logged though, just 404s. You can confirm by trying to generate a playbook with just punctuation as input. You'll see the response body isn't logged anywhere.

Reply:
@robinbobbitt I now [log ](https://github.com/ansible/ansible-ai-connect-service/pull/1157/files#diff-1705f63f7b89989770c6c1a60b68c2857dd8729e8d2fee5024eaef42dab3a089R48-R56)the details if _any_ exception thrown by our response handling; or would you rather just log it for 404's and 400's?

Reply:
Nice! That seems good to me. Thank you.